### PR TITLE
All fa-check and IconCheck component changed to Icon component

### LINKED ISSUE #17526: Replace all fa-check icons with equivalent Icon component
### Description
In an effort to replace all font-awesome icons with the new `Icon` component it would be great to replace all `fa-check` and `<IconCheck />` components with the `Icon` equivalent

One way to find all the `fa-check` icons would be to search "fa-check"

Screenshot 2023-01-31 at 1 18 02 PM

Same with `<IconCheck`

Screenshot 2023-01-31 at 1 20 28 PM

### Technical Details
- Replace all `fa-check` font awesome icons with `<Icon name={ICON_NAMES.CHECK} />`
- Replace all `<IconCheck` font awesome icons with `<Icon name={ICON_NAMES.CHECK} />`
- Use `import { Icon, ICON_NAMES } from '../../component-library'` path will be subject to change depending on where it is being imported from 

### Acceptance Criteria 
- All `fa-check` have been replaced with their `<Icon name={ICON_NAMES.CHECK} />` component equivalent
- All `<IconCheck />` have been replaced with their `<Icon name={ICON_NAMES.CHECK} />` component equivalent
- PR includes screenshot of before and after of all instances can do it in the code comments if it makes more sense

### References
- [**Icon component documentation in storybook**](https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-icon--default-story#icon)

Screenshot 2023-01-31 at 1 21 36 PM

### PR DESCRIPTION
**Pull request by spiritanand**
Fixes #17526 

## Explanation
We are using multiple icons for a check mark. We need to change it for uniformity across the board.
I have updated all the check icons to our Icon component.

### Before

image
image
image
image

### After

image
image
image
image

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by spiritanand** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by NidhiKJha — COMMENTED** [review]
Thanks for this PR, really great start. Left a few suggestions. Also, it will be great if you can resolve the conflicts and add screenshots for the changes. It will help the reviewers to review it 😄

File: ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js [hunk 1]
```diff
@@ -72,17 +73,25 @@ const NetworksListItem = ({
...
+        </>
       ) : (
-        <IconCheck
-          className="networks-tab__content__icon-check"
-          color="transparent"
-          aria-hidden="true"
-        />
+        <>
+          <h1>Network List</h1>
+          <Icon
```
[the remark this answers is not in the document]
Comment (spiritanand) [reply ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js:88]: Yes, that is ny bad. I was using it for testing. I will resolve it.

File: ui/pages/first-time-flow/create-password/new-account/new-account.component.js [hunk 2]
```diff
@@ -192,7 +194,11 @@ export default class NewAccount extends PureComponent {
               aria-checked={termsChecked}
               aria-labelledby="ftf-chk1-label"
             >
-              {termsChecked ? <i className="fa fa-check fa-2x" /> : null}
+              {termsChecked ? (
+                <>
+                  <Icon name={ICON_NAMES.CHECK} size={SIZES.XL} />
+                </>
+              ) : null}
```
[the remark this answers is not in the document]
Comment (spiritanand) [reply ui/pages/first-time-flow/create-password/new-account/new-account.component.js:201]: Ok sure.

**Comment by spiritanand** [comment]
> Thanks for this PR, really great start. Left a few suggestions. Also, it will be great if you can resolve the conflicts and add screenshots for the changes. It will help the reviewers to review it 😄

I have added the screenshots of as many components as I could find in the storybook and flow of using the extension.

File: yarn.lock [hunk 3]
```diff
@@ -12338,7 +12338,7 @@ __metadata:
   dependencies:
     inherits: ^2.0.3
     readable-stream: ^3.0.2
-  checksum: 1cef636e7061f310088706b34fe774e3960dff60a5039158b5e5c84795f6dd8a3411659324280405b8c5f1d7e8e3d4f68fa48e55963ed14953a44fef66423329
+  checksum: cae463db86ffa444ad60051a4a6c17e3f3d289fd169a87cbf1a6ee9d4118aafb976bb4713943359dba9c8211e2250f507bc4d50659b383e6b4e8dc1f0320e548
```
[the remark this answers is not in the document]
Comment (spiritanand) [reply yarn.lock:12341]: Yeah, but this is an auto generated file. I did not edit it maually.

**Comment by spiritanand** [comment]
I cannot understand what is creating the conflicts in my branch.

**Comment by georgewrmarshall** [comment]
Hey @spiritanand, I think it was this PR unfortunately it got merged before yours. https://github.com/MetaMask/metamask-extension/commit/794f3039f187c1ce6230d1206f109c638daa2cac it updates a lot of the design-system constants. I think once you update those instances you should be goof!

**Comment by spiritanand** [comment]
@georgewrmarshall I am really confused on how to get all the changes from the updated develop branch to my own branch. `git pull upstream` did not update it. Really confused.

**Pull request closed without merging**